src/com/dotcms/rest/BaseRestPortlet.java

```diff
@@ -184,24 +187,74 @@ public Response getLayout ( @Context HttpServletRequest request, @Context HttpSe
 
 	}
 
-	private class ResponseWrapper extends HttpServletResponseWrapper {
-
-		private StringWriter writer = new StringWriter();
+	public class ResponseWrapper extends HttpServletResponseWrapper{
+		private ByteArrayOutputStream output;
+		private StringWriter writer;
+		private int contentLength;
+		private String contentType;
 
 		public ResponseWrapper(HttpServletResponse response) {
 			super(response);
+			output = new ByteArrayOutputStream();
+			writer = new StringWriter();
+		}
+
+		public ServletOutputStream getOutputStream() {
+			return new FilterServletOutputStream(output);
+		}
+
+		public byte[] getData() {
+			return output.toByteArray();
 		}
 
 		public String getResponseString() {
-			return writer.toString();
+			return UtilMethods.isSet(writer.toString()) ? writer.toString() : output.toString();
```

Comment:
Prefer StringUtils#empty/notEmpty over UtilMethods.isSet. Although in this case testing on 'writer.getBuffer().length() > 0' would be even better.